metroae

```diff
@@ -1644,7 +1653,19 @@ function download_container_templates {
         write_to_screen_and_script_log ""
         exit 1
     fi
-    curl -O $TEMPLATE_TAR_LOCATION
+    {
+        curl -O $TEMPLATE_TAR_LOCATION &> /dev/null
+    } || {
+        echo "Warning: Internet connection not detected. Template files were not downloaded"
+        echo "Please download the templates manually from the following URL:"
+        echo "$TEMPLATE_TAR_LOCATION"
+        echo ""
+        echo "Copy the VSD specifications from this URL"
+        echo "$VSD_SPECIFICATIONS_LOCATION"
+        echo "Upon successful download of the two tar files"
+        echo "untar the files to: $METROAE_MOUNT_POINT"
+        exit 0
```

Comment:
Do not exit 0 here. Let the script continue to do other things

Reply:
if there isn't a connection and I don't exit 0, won't the script attempt to download the vsd specifications using curl, which will cause an error? The section below acts as a try-catch to see if the first curl succeeds, and gives a warning and exits if it doesn't (due to a connection error):

```
    {
        curl -O $TEMPLATE_TAR_LOCATION &> /dev/null
    } || {
        echo "Warning: Internet connection not detected. Template files were not downloaded"
        echo "Please download the templates manually from the following URL:"
        echo "$TEMPLATE_TAR_LOCATION"
        echo ""
        echo "Copy the VSD specifications from this URL"
        echo "$VSD_SPECIFICATIONS_LOCATION"
        echo "Upon successful download of the two tar files"
        echo "untar the files to: $METROAE_MOUNT_POINT"
        exit 0
    }
```

At the end of the download_container_templates function there was already an exit 0 (although I've moved some things around, I haven't moved that particular exit 0's location), but it only gets to the end of the function if all the downloading (via curl) of the tar files and the untaring was successful.

I don't fully understand why I shouldn't exit 0 at that spot.

Reply:
Okay, so your approach is to fail if one of the links fails and not try the other link. Okay.

Reply:
ah okay, I think I understand what you mean now. I'll add both curls to the try block. If either curl fails, then it goes to the catch block